pandas/core/indexes/datetimelike.py

```diff
@@ -430,6 +430,10 @@ def min(self, axis=None, *args, **kwargs):
         --------
         numpy.ndarray.min
         """
+        if axis is not None and axis >= self.ndim:
+            raise ValueError("`axis` must be fewer than the number of "
```

Comment:
don't do this here, rather this should be in valididate_* functions (if you think this is really necessary and you have a test for it)